src/RealtimeEffectPanel.cpp

```diff
@@ -987,6 +971,35 @@ RealtimeEffectPanel::RealtimeEffectPanel(wxWindow* parent, wxWindowID id, const
    mEffectList = effectList;
 
    SetSizerAndFit(vSizer.release());
+
+   mTrackListChanged = TrackList::Get(mProject).Subscribe([this](const TrackListEvent& evt) {
+         auto track = evt.mpTrack.lock();
+         auto waveTrack = dynamic_cast<WaveTrack*>(track.get());
+
+         if (waveTrack == nullptr)
+            return;
+         
+         switch (evt.mType)
+         {
+         case TrackListEvent::TRACK_DATA_CHANGE:
+            if (mCurrentTrack.lock() == track)
```

Comment:
I think this may fail to fire when intended, in case `mCurrentTrack` expired because the track was replaced because some other edit was done to it that caused TrackList::Replace

Reply:
I have checked and mCurrentTrack is always consistent

Reply:
I have also